src/taipy/gui/utils/proxy.py

```diff
@@ -93,10 +94,16 @@ def __init__(self, gui: "Gui", listening_port: int) -> None:
     def run(self):
         if self._is_running:
             return
-        self._is_running = True
-        site = Site(_TaipyReverseProxyResource(self._gui._get_config("host", "127.0.0.1"), b"", self._gui))
-        reactor.listenTCP(self._listening_port, site)
+        host = self._gui._get_config("host", "127.0.0.1")
+        port = self._listening_port
+        if _is_port_open(host, port):
+            raise ConnectionError(
+                f"Port {port} is already opened on {host}. You have another server application running on the same port."
```

Comment:
So that is going to be handy.

Reply:
Indeed. You found the bug while playing around with it 😀